File: C++/1 Two Sum.cpp
using namespace std;

class Solution
{
public:
    vector<int> twoSum(vector<int> &nums, int target)
    {
        unordered_map<int, int> seen;
        int n = nums.size();
        for (int i = 0; i < n; i++)
        {
            int x = target - nums[i];
            if (seen.find(x) != seen.end())
            {
                return {i, seen[x]};
            }
            seen[nums[i]] = i;
        }
        return {-1, -1};
    }
};

File: Swift/1.Two sum.swift
class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        var seen: [Int: Int] = [:]
        for (idx, val) in nums.enumerated() {
            let y = target - val
            let next = seen[y]
            if let j = next {
                return [j, idx]
            }
            seen[val] = idx
        }
        return [-1, -1]
    }
}
